### PR TITLE
Table: Component progress & custom FieldConfig options

### PR DESCRIPTION
**Pull request by torkelo**
* Change to useFlexLayout  (cells can expand, you set minWidth) 
* flex layout did not support fixed size columns, so changed back to block layout and smarter width allocation, so some columns can have defined width and the others take the rest. Can be made more dynamic (percentage based) in the future. 
* Most naive way to pass custom field options via custom prop on FieldConfig (width & text align)
*

### CODE REVIEW

File: packages/grafana-data/src/field/fieldOverrides.ts [hunk 1]
```diff
@@ -77,7 +77,7 @@ export function applyFieldOverrides(options: ApplyFieldOverrideOptions): DataFra
       const info = fieldMatchers.get(rule.matcher.id);
       if (info) {
         override.push({
-          match: info.get(rule.matcher),
+          match: info.get(rule.matcher.options),
```
Comment (torkelo) [inline packages/grafana-data/src/field/fieldOverrides.ts]: found this bug when starting using byName field matcher

File: public/app/plugins/datasource/testdata/StreamHandler.ts [hunk 2]
```diff
@@ -1,519 +0,0 @@
...
-  spread: 3.5,
-  noise: 2.2,
-  bands: 1,
-};
-
-type StreamWorkers = {
-  [key: string]: StreamWorker;
-};
-
-export class StreamHandler {
```
Comment (ryantxu) [inline public/app/plugins/datasource/testdata/StreamHandler.ts]: See: 
https://github.com/grafana/grafana/pull/21234

File: packages/grafana-data/src/field/fieldOverrides.ts (hunk 1 above)
Comment (ryantxu) [reply packages/grafana-data/src/field/fieldOverrides.ts]: yup -- 👍

**Review by ryantxu — APPROVED** [review]
This is looking great -- I have some ideas on how to approach configuring custom fields, but would like to do that after this is merged.

Given that this is mostly changing an alpha panel with components that are not (yet!) used in production, I think we should merge quickly and refine with follow on PRs.

File: public/app/plugins/panel/table/types.ts [hunk 3]
```diff
@@ -18,3 +17,26 @@ export interface TableRenderModel {
   columns: ColumnRender[];
   rows: any[][];
 }
+
+export interface ColumnStyle {
```
Comment (ryantxu) [inline public/app/plugins/panel/table/types.ts]: Eventually IIUC, this should all be represented with FieldConfig + custom.

Comment (ryantxu) [reply public/app/plugins/panel/table/types.ts]: Nevermind -- I see this is on the old table plugin!

File: packages/grafana-ui/src/components/BarGauge/BarGauge.tsx [hunk 4]
```diff
@@ -39,22 +39,30 @@ export interface Props extends Themeable {
   minValue: number;
   orientation: VizOrientation;
   itemSpacing?: number;
-  displayMode: 'basic' | 'lcd' | 'gradient';
+  cellWidth?: number;
```
Comment (ryantxu) [inline packages/grafana-ui/src/components/BarGauge/BarGauge.tsx]: cellWidth is a non-obvious name/meaning for BarGauge.

If it needs to be an external property, can we add comments about what it is supposed to do?

File: packages/grafana-ui/src/components/Table/TableCellBuilder.tsx [hunk 5]
```diff
@@ -1,324 +0,0 @@
...
-
-  return (
-    <div style={style} className={'gf-table-cell ' + className}>
-      {value}
-    </div>
-  );
-};
-
-// ***************************************************************************
-// HERE BE DRAGONS!!!
```
Comment (ryantxu) [inline packages/grafana-ui/src/components/Table/TableCellBuilder.tsx]: Good riddance!

**Comment by ryantxu** [comment]
ugg -- sorry, I merged with master not realizing this is branched off `new-table-display-processor`  now my git foo has reached its limits!

File: packages/grafana-ui/src/components/BarGauge/BarGauge.tsx (hunk 4 above)
Comment (torkelo) [reply packages/grafana-ui/src/components/BarGauge/BarGauge.tsx]: yea, can rename to lcdCellWidth

**Pull request merged**